modules/aws/irsa/main.tf

```diff
@@ -53,11 +53,27 @@ resource "aws_iam_role" "this" {
 }
 
 resource "aws_iam_policy" "permissions" {
+  for_each = {
+    for s in ["policy-attachment"] :
+    s => s
+    if var.policy_json != ""
+  }
   name   = var.name
   policy = var.policy_json
 }
 
 resource "aws_iam_role_policy_attachment" "permissions" {
-  policy_arn = aws_iam_policy.permissions.arn
+  for_each = {
+    for s in ["policy-attachment"] :
+    s => s
+    if var.policy_json != ""
+  }
+  policy_arn = aws_iam_policy.permissions[0].arn
```

Comment:
`policy_arn = aws_iam_policy.permissions["policy-attachment"].arn` would make more sense to me but an index might work as well, I don't know...

Reply:
Agreed, I have a bad habit of using index instead.